src/frontend/js/hooks/useCourseProductUnion/index.ts

```diff
@@ -0,0 +1,51 @@
+import { defineMessages } from 'react-intl';
+import { useJoanieApi } from 'contexts/JoanieApiContext';
+import {
+  CourseListItem,
+  Product,
+  PaginatedResourceQuery,
+  CourseProductRelation,
+} from 'types/Joanie';
+import useUnionResource from 'hooks/useUnionResource';
+
+export const isCourseListItem = (obj: CourseListItem | Product): obj is CourseListItem => {
+  return 'course_runs' in obj;
+};
+export const isProduct = (obj: CourseListItem | Product): obj is Product => {
+  return 'certificate_definition' in obj;
+};
+
+const messages = defineMessages({
+  errorGet: {
+    id: 'hooks.useCourseProductUnion.errorGet',
+    description: 'Error message shown to the user when courses and products fetch request fails.',
+    defaultMessage: 'An error occurred while fetching courses and products. Please retry later.',
```

Comment:
WDYT to use the term `trainings` to merge notion of course and course product relations ?
```suggestion
    description: 'Error message shown to the user when courses and course product relations fetch request fails.',
    defaultMessage: 'An error occurred while fetching trainings. Please retry later.',
```

Reply:
I feel like a course product relation is a training and a course is a part of a training.
But i feel like we need something to named them both and i don't have a better word.
Then why not.